addons/web_editor/tools.py

```diff
@@ -0,0 +1,132 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import re
+import requests
+import odoo
+
+from odoo import _
+from odoo.tools import image_process
+from odoo.modules.module import get_module_resource
+
+
+def get_video_platform_and_id(video_url):
+    ''' Computes the valid source and document ID from given URL
+        (or False in case of invalid URL).
+    '''
+    if not video_url:
+        return False
+
+    # To detect if we have a valid URL or not
+    valid_url_regex = r'^(http:\/\/|https:\/\/|\/\/)[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$'
+
+    # Regex for few of the widely used video hosting services
+    youtube_regex = r'^(?:(?:https?:)?\/\/)?(?:www\.)?(?:youtu\.be\/|youtube(-nocookie)?\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((?:\w|-){11})(?:\S+)?$'
+    vimeo_regex = r'\/\/(player.)?vimeo.com\/([a-z]*\/)*([0-9]{6,11})[?]?.*'
+    vine_regex = r'\/\/vine.co\/v\/(.[a-zA-Z0-9]*)'
+    dailymotion_regex = r'(?:.+dailymotion.com\/(video|hub|embed\/video|embed)|dai\.ly)\/([^_?]+)[^#]*(#video=([^_&]+))?'
+    instagram_regex = r'(?:(.*)instagram.com|instagr\.am)\/p\/(.[a-zA-Z0-9-_\.]*)'
+    youku_regex = r'(.*).youku\.com\/(v_show\/id_|embed\/)(.+)'
```

Comment:
can we move it outside to be callable from outside if needed. 
(even if it is probably wrong idea, it will avoid specific case to copy past these regexp again and again)

They will be able to just call:
otoo.tools.players_external_regexp['youtube']

Reply:
> can we move it outside to be callable from outside if needed.
> (even if it is probably wrong idea, it will avoid specific case to copy past these regexp again and again)
> 
> They will be able to just call:
> otoo.tools.players_external_regexp['youtube']

can we move all the stuff inside odoo.tools.players_utils.py(new file)?? and 
move this route in the web controller for fronted use: https://github.com/odoo/odoo/pull/44537/files#diff-a30f997f5c64ebc472e894fe6b6de476R158 ??

Reply:
while we only use it in web_Editor module, or apps that's depends of this last (slide, website, ...) I will keep it here... Except if you see another case that I'm missing...?

Imo, just need to define,it outside of the function to be reusable for now.

Thanks